src/sql/workbench/contrib/backup/browser/backup.component.ts

```diff
@@ -528,19 +537,22 @@ export class BackupComponent extends AngularDisposable {
 			this.recoveryBox!.disable();
 			this.mediaNameBox!.disable();
 			this.mediaDescriptionBox!.disable();
-			if (this._engineEdition === DatabaseEngineEdition.SqlManagedInstance) {
+			if (this._allowFileBackup && this._allowUrlBackup) {
+				this.toUrlCheckBox.enable();
+			}
+			if (this._allowUrlBackup) {
 				this.toUrlCheckBox.checked = true;
 				this.copyOnlyCheckBox.checked = true;
-				this.backupTypeSelectBox!.disable();
-				this.copyOnlyCheckBox!.disable();
-				this.backupRetainDaysBox!.disable();
-				this.disableMedia = true;
-			} else {
-				this.toUrlCheckBox.checked = false;
-				this.copyOnlyCheckBox.checked = false;
+			}
+			if (this._allowFileBackup) {
+				this.backupTypeSelectBox!.enable();
 				this.backupRetainDaysBox!.enable();
 				this.copyOnlyCheckBox!.enable();
-				this.backupTypeSelectBox!.enable();
+
+				this.disableMedia = false;
+			} else {
+				this.backupRetainDaysBox!.enable();
```

Comment:
Should this be disabled instead to match the MI section from before? Also, should the backupType select box also be disabled here, since this section is for URL-only mode?